src/query/api/v1/handler/prometheus/native/read.go

```diff
@@ -135,8 +136,16 @@ func (h *promReadHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	handleroptions.AddWarningHeaders(w, result.Meta)
 	h.promReadMetrics.fetchSuccess.Inc(1)
 
+	keepNans := h.opts.Config().ResultOptions.KeepNans
+	switch result.Meta.KeepNans {
+		case block.BoolFalse:
+			keepNans = false
+		case block.BoolTrue:
+			keepNans = true
+	}
```

Comment:
Don't think we'd ever want to overwrite the case where the config keeps Nans, so this can be simplified after making `result.Meta.KeepNans` an actual bool

```suggestion
	keepNans := h.opts.Config().ResultOptions.KeepNans
	if !keepNans {
		keepNans = result.Meta.KeepNans 
	}
```

Also, can we rename `Nans` to `NaNs` within codebase? Probably fine to leave that to another PR though